cmd/clusterctl/cmd/config_cluster.go

```diff
@@ -65,50 +73,110 @@ var configClusterClusterCmd = &cobra.Command{
+	// flags for the template variables
 	configClusterClusterCmd.Flags().StringVarP(&cc.targetNamespace, "target-namespace", "n", "", "The namespace where the objects describing the workload cluster should be deployed. If not specified, the current namespace will be used")
 	configClusterClusterCmd.Flags().StringVarP(&cc.kubernetesVersion, "kubernetes-version", "", "", "The Kubernetes version to use for the workload cluster. By default (empty), the value from os env variables or the .cluster-api/clusterctl.yaml config file will be used")
 	configClusterClusterCmd.Flags().IntVarP(&cc.controlPlaneMachineCount, "control-plane-machine-count", "", 1, "The number of control plane machines to be added to the workload cluster.")
 	configClusterClusterCmd.Flags().IntVarP(&cc.workerMachineCount, "worker-machine-count", "", 0, "The number of worker machines to be added to the workload cluster.")
 
+	// flags for the repository source
+	configClusterClusterCmd.Flags().StringVarP(&cc.infrastructureProvider, "infrastructure", "i", "", "The infrastructure provider to read the workload cluster template from. By default (empty), the default infrastructure provider will be used if no other source is specified")
+	configClusterClusterCmd.Flags().StringVarP(&cc.flavor, "flavor", "f", "", "The workload cluster template variant to be used when reading from the infrastructure provider repository. By default (empty), the default cluster template will be used")
+
+	// flags for the url source
+	configClusterClusterCmd.Flags().StringVarP(&cc.URL, "from", "", "", "The URL to read the workload cluster template from. By default (empty), the infrastructure provider repository URL will be used")
+
+	// flags for the config map source
+	configClusterClusterCmd.Flags().StringVarP(&cc.configMapName, "from-config-map", "", "", "The ConfigMap to read the workload cluster template from. This can be used as alternative to read from the provider repository or from an URL")
+	configClusterClusterCmd.Flags().StringVarP(&cc.configMapNamespace, "from-config-map-namespace", "", "", "The namespace where the ConfigMap exists. By default (empty), the current namespace will be used")
+	configClusterClusterCmd.Flags().StringVarP(&cc.configMapDataKey, "from-config-map-key", "", "", fmt.Sprintf("The ConfigMap.Data key where the workload cluster template is hosted. By default (empty), %q will be used", client.DefaultCustomTemplateConfigMapKey))
+
+	// other flags
+	configClusterClusterCmd.Flags().BoolVarP(&cc.listVariables, "list-variables", "", false, "Returns the list of variables expected by the template instead of the template yaml")
+
 	configCmd.AddCommand(configClusterClusterCmd)
 }
 
-func runGenerateCluster(name string) error {
+func runGetClusterTemplate(name string) error {
 	c, err := client.New(cfgFile)
 	if err != nil {
 		return err
 	}
 
-	options := client.GetClusterTemplateOptions{
+	templateOptions := client.GetClusterTemplateOptions{
```

Comment:
Just a preference but we could just call this object `client.ClusterTemplateOptions` unless there's some conflict elsewhere.

Reply:
All the options object are called like the corresponding action, so I prefer to keep this consistent